### PR TITLE
Complete airport connections entity. See #55.

### PR DESCRIPTION
**Pull request by ChrisHng**

### CODE REVIEW

File: docroot/modules/custom/dct_airport_connections/src/AirportConnectionsAccessControlHandler.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+namespace Drupal\dct_airport_connections;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Entity\EntityAccessControlHandler;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Session\AccountInterface;
+
+/**
+ * Class AirportConnectionsAccessControlHandler
```
Comment (lauracernau) [inline docroot/modules/custom/dct_airport_connections/src/AirportConnectionsAccessControlHandler.php]: The comment should say what is this class for.

File: docroot/modules/custom/dct_airport_connections/src/AirportConnectionsListBuilder.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Drupal\dct_airport_connections;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityListBuilder;
+
+/**
+ * Class AirportConnectionsListBuilder
```
Comment (lauracernau) [inline docroot/modules/custom/dct_airport_connections/src/AirportConnectionsListBuilder.php]: The comment should say what is this class for.

File: docroot/modules/custom/dct_airport_connections/src/Entity/AirportConnections.php [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+  public function getChangedTime() {
+    return $this->get('changed')->value;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setChangedTime($timestamp) {
+    $this->set('changed', $timestamp);
+    return $this;
```
Comment (lauracernau) [inline docroot/modules/custom/dct_airport_connections/src/Entity/AirportConnections.php]: Here should be a newline

File: docroot/modules/custom/dct_airport_connections/src/Entity/AirportConnections.php [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+  /**
+   * {@inheritdoc}
+   */
+  public function getChangedTimeAcrossTranslations()  {
+    $changed = $this->getUntranslated()->getChangedTime();
+    foreach ($this->getTranslationLanguages(FALSE) as $language)    {
+      $translation_changed = $this->getTranslation($language->getId())->getChangedTime();
+      $changed = max($translation_changed, $changed);
+    }
+    return $changed;
```
Comment (lauracernau) [inline docroot/modules/custom/dct_airport_connections/src/Entity/AirportConnections.php]: Here should be a newline.

File: docroot/modules/custom/dct_airport_connections/src/Entity/AirportConnections.php [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+  public function getOwnerId() {
+    return $this->get('user_id')->target_id;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setOwnerId($uid) {
+    $this->set('user_id', $uid);
+    return $this;
```
Comment (lauracernau) [inline docroot/modules/custom/dct_airport_connections/src/Entity/AirportConnections.php]: Here should be a newline.

File: docroot/modules/custom/dct_airport_connections/src/Entity/AirportConnections.php [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setOwner(UserInterface $account) {
+    $this->set('user_id', $account->id());
+    return $this;
```
Comment (lauracernau) [inline docroot/modules/custom/dct_airport_connections/src/Entity/AirportConnections.php]: Here should be a newline.

File: docroot/modules/custom/dct_airport_connections/src/Entity/AirportConnections.php [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+ *     "delete-form" = "/admin/airport-connections/{airport_connections}/delete",
+ *     "collection" = "/admin/airport-connections/list"
+ *   },
+ *   field_ui_base_route = "airport_connections.settings",
+ * )
+ */
+class AirportConnections extends ContentEntityBase implements AirportConnectionsInterface {
+
+  /**
+   * {@ingeritdoc}
```
Comment (corneaalina) [inline docroot/modules/custom/dct_airport_connections/src/Entity/AirportConnections.php]: Typo on 'ingeritdoc'

File: docroot/modules/custom/dct_airport_connections/src/Form/AirportConnectionsSettingsForm.php [hunk 8]
```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Drupal\dct_airport_connections\Form;
+
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Class AirportConnectionsSettingsForm
```
Comment (lauracernau) [inline docroot/modules/custom/dct_airport_connections/src/Form/AirportConnectionsSettingsForm.php]: The comment should say what is this class for.

File: docroot/modules/custom/dct_airport_connections/src/DataProvider.php [hunk 9]
```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Drupal\dct_airport_connections;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+
+/**
+ * Class DataProvider
```
Comment (lauracernau) [inline docroot/modules/custom/dct_airport_connections/src/DataProvider.php]: The comment should say what is this class for.

File: docroot/modules/custom/dct_airport_connections/src/DataProvider.php [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+        // Visual customization of the link.
+        'factor' => 0.3,
+        'attrs' => [
+          'stroke-width' => 2,
+        ],
+      ];
+    }
+
+    return $links;
+  }
```
Comment (corneaalina) [inline docroot/modules/custom/dct_airport_connections/src/DataProvider.php]: A newline should be here.

File: docroot/modules/custom/dct_airport_connections/src/DataProviderInterface.php [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+   * @return array
+   */
+  public function getPlots();
+
+  /**
+   * Returns an array containing the links between the home city and the connections.
+   *
+   * @return array
+   */
+  public function getLinks();
```
Comment (corneaalina) [inline docroot/modules/custom/dct_airport_connections/src/DataProviderInterface.php]: A newline should be here.

File: docroot/modules/custom/dct_airport_connections/src/Entity/AirportConnections.php [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getChangedTimeAcrossTranslations()  {
+    $changed = $this->getUntranslated()->getChangedTime();
+    foreach ($this->getTranslationLanguages(FALSE) as $language)    {
+      $translation_changed = $this->getTranslation($language->getId())->getChangedTime();
+      $changed = max($translation_changed, $changed);
+    }
```
Comment (corneaalina) [inline docroot/modules/custom/dct_airport_connections/src/Entity/AirportConnections.php]: Newline here.

File: docroot/modules/custom/dct_airport_connections/dct_airport_connections.info.yml [hunk 13]
```diff
@@ -0,0 +1,5 @@
+name: DCT Airport Connections
```
Comment (corneaalina) [inline docroot/modules/custom/dct_airport_connections/dct_airport_connections.info.yml]: Configurations missing.

File: docroot/modules/custom/dct_airport_connections/dct_airport_connections.permissions.yml [hunk 14]
```diff
@@ -0,0 +1,11 @@
+view airport_connections entity:
```
Comment (danielpopdan) [inline docroot/modules/custom/dct_airport_connections/dct_airport_connections.permissions.yml]: The entity shouldn't have a detail page.

**Pull request closed without merging**